valqueries-test/src/test/java/com/valqueries/automapper/AutoMapperIT.java

```diff
@@ -313,10 +308,32 @@ public void eagerLoad_limits() throws Throwable {
 		assertEquals(2 * (2 + 2), (long)carsFound.stream().reduce(0, (accumulator, car) -> accumulator + car.getDoors().size() + car.getWheels().size(), Integer::sum));
 	}
 
-	private void carWithDoorsAndWheels() {
+	@Test
+	@TestClasses({Car.class, Door.class, CarWheel.class})
+	public void sorting_happy() throws Throwable {
+
+		List<String> carTitles = Arrays.asList("C","B","A");
+		carTitles.forEach(carTitle-> carWithDoorsAndWheels(carTitle));
+
+		List<Car> carsFound = carRepository.query()
+				.subQueryList(Car::getDoors, sq -> {
+					sq.in(Door::getTitle, "Nissan door 1");
+				})
+				.sortAscending(Car::setTitle)
+				.execute().collect(Collectors.toList());
+
+		Collections.sort(carTitles);
+
+		Iterator<Car> carsFoundIter = carsFound.iterator();
+
+		carTitles.forEach(carTitle-> assertEquals(carTitle,carsFoundIter.next().getTitle()));
```

Comment:
can't you compare `carTitles` with the collected titles from db instead of doing an assertion on 1 by 1?